### PR TITLE
make pepserver a python package

### PR DESCRIPTION
**Pull request by stolarczyk**
multiple changes:

- make a Python pkg
- add v1 router
- setup api versioning
- add CLI argparser

---

@NLeRoy917, I have set up the server following the refgenieserver example. Now you can do this:

```
# in the repo
pip install .
pepserver serve --port 8000
```

feel free to change stuff and lmk if something doesn't make sense!

### CODE REVIEW

**Comment by nleroy917** [comment]
Moved a const and changed the year in the license.

I think now that I am more comfortable with FastAPI and uvicorn it's making sense. Where is the entrypoint for the docker container? I see the installation of the package inside the docker container. And I believe I see the python package entrypoint:

```python
# setup.py
...
    entry_points={
        "console_scripts": [
            "{p} = {p}.__main__:main".format(p=PACKAGE),
        ],
    },
...
```

But how does Docker know to run `pepserver serve --port 8000` ? It ends at 
```docker
RUN pip install .
```
Also, just curious, what are the benefits to setting it up as a python package v what the [FastAPI](https://fastapi.tiangolo.com/deployment/docker/) docs lay out?

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,5 @@
...
+FROM tiangolo/uvicorn-gunicorn:python3.7-alpine3.8
+LABEL authors="Nathan LeRoy, Michal Stolarczyk, Nathan Sheffield"
 
-# dependencies
-COPY requirements.txt /
-RUN pip install  pip
-RUN pip install -r /requirements.txt
\ No newline at end of file
+COPY . /app
+RUN pip install .
```
Comment (nleroy917) [inline Dockerfile:5]: How does docker know what to do after this?

**Review by nleroy917 — APPROVED** [review]
Looks good! Thanks for the additional help. I think I'm following here. Just mostly confused about deployment & docker setup at this point...

**Comment by stolarczyk** [comment]
> But how does Docker know to run pepserver serve --port 8000 ?

This image will need to be imported in another one that will have a docker entrypoint directive, like this one: https://github.com/refgenie/aws_igenomes/blob/1d935217ce7eca43dbf30cb6a46a90a010313036/Dockerfiles/aws_igenomes.Dockerfile#L5

Setting this up this way separates software from data/metadata served.

> Also, just curious, what are the benefits to setting it up as a python package v what the FastAPI docs lay out?

hmm.. more standardized software distribution and versioning? Both approaches are fine I guess. But setting this up this way will make things easier down the road as we will be able to follow what has been done for other servers :)

**Comment by nleroy917** [comment]
Oh! I was unaware we'd import this image into another. All makes perfect sense now...

**Pull request merged**